gdquest-godot-theme/addons/rich_text_editor/RichTextEditor.gd

```diff
@@ -0,0 +1,154 @@
+tool
+extends EditorPlugin
+class_name RichTextPlugin
+
+signal text_edit_popped(new_text_edit)
+
+const BUTTON_SCENE: PackedScene = preload("res://addons/rich_text_editor/interface/Button.tscn")
+const INTERFACE_SCENE: PackedScene = preload("res://addons/rich_text_editor/interface/RichEditorInterface.tscn")
+
+var rich_labels: Array = []
+var text_edit: TextEdit = TextEdit.new()
+
+var _editor: = get_editor_interface()
+var _button: Button = BUTTON_SCENE.instance()
+var _selection: = _editor.get_selection()
+var _interface: RichEditorInterface
+
+func _ready() -> void:
+	_button.hide()
+	_button.connect("pressed", self, "_on_Button_pressed")
+	
+	if not _editor.is_plugin_enabled("gdquest_docker"):
+		add_control_to_container(CONTAINER_CANVAS_EDITOR_MENU, _button)
+	
+	_selection.connect("selection_changed", self, "_on_EditorSelection_selection_changed")
+
+
+func _exit_tree() -> void:
+	if not _editor.is_plugin_enabled("gdquest_docker"):
+		remove_control_from_container(CONTAINER_CANVAS_EDITOR_MENU, _button)
+
+
+func _on_Button_pressed() -> void:
+	initialize_interface()
+
+
+func _on_TextEdit_tree_exited() -> void:
+	_button.disabled = false
+
+
+func _on_TextEdit_text_changed() -> void:
+	var undo: = get_undo_redo()
+	
+	undo.create_action("Set BBCode Text")
+	for rich_text_label in rich_labels:
+		rich_text_label.bbcode_enabled = true
+		undo.add_undo_property(rich_text_label, "bbcode_text", rich_text_label.bbcode_text)
+		undo.add_do_property(rich_text_label, "bbcode_text", text_edit.text)
```

Comment:
I may change this to a symmetric approach, swaping line `48` with `49` and making it similar to line `47` to prevent confusions